prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -496,24 +518,28 @@ class HttpClientUtils {
             httpClient: HttpClient? = null,
         ): HttpResponse {
             return runBlocking {
-                (httpClient ?: createDefaultHttpClient(accessToken)).request(url) {
+                (httpClient ?: getDefaultHttpClient()).request(url) {
                     this.method = method
                     timeout {
                         requestTimeoutMillis = timeout
                     }
                     url {
                         queryParameters?.forEach {
-                            parameter(it.key, it.value.toString())
+                            parameter(it.key, it.value)
                         }
                     }
-
                     headers?.let {
                         headers {
                             headers.forEach {
                                 append(it.key, it.value)
                             }
                         }
                     }
+                    accessToken?.let {
+                        headers {
+                            append("Authorization", "Bearer $accessToken")
```

Comment:
Could there be another type of auth needed in the future outside of `Bearer`?

Reply:
not in the current design no. 

@arnejduranovic mentioned something about - going forward - each microservice is going to have it's own client for managing this stuff. I imagine in that case if client [x] is doing auth differently than [y] we could make those changes locally to each microservice.